### ISSUE TITLE
Modified email templates are ignored by the system

### ISSUE THREAD

**Issue by LolaCapitani** [chamilo/chamilo-lms#3380]
Hi!
I found some (but probably not all) email templates that receive students / teachers / administrators in the path: 
"main/template/default/mail/file_name.tpl".

So I need to modify some email templates (change variables; start a new paragraph, etc.).

For example, I need to modify the file:
- new_user_first_email_confirmation.tpl

I followed the instructions written in the README.md file in the "overrides" folder (main/template/overrides).

I then created inside the "overrides" folder the "mail" folder (photo 01).

I modified the file new_user_first_email_confirmation.tpl

Inside the "mail" folder I then copied the modified file new_user_first_email_confirmation.tpl (photo 02)

![01](https://user-images.githubusercontent.com/67590937/87676437-e928f580-c778-11ea-9bc0-97d5239752f0.jpg)
![02](https://user-images.githubusercontent.com/67590937/87676444-e9c18c00-c778-11ea-9ec9-75e189ce4be8.jpg)

The problem is that when the user receives the email (in this case it is the first registration email), the template is not the one I modified. Inside in fact there are the errors present in the original template.

I also tried with a clean installation, but I have the same problem.

Is there something wrong?

Thanks for your work!

Lola
(=^ ◡ ^=)

-----------------------------------------------------------------------------
Desktop 

    OS: Linux Ubuntu 18.04 / Windows 10(64bit), Win 7(32 bit)
    Browser CHROME 80.0.3987.87 , FIREFOX 78.01

Server

    OS: Linux 3.10.0, LiteSpeed V7.6,
    Version of Chamilo: 1.11.10
    Version of PHP: PHP 7.1

-----------------------------------------------------------------------------

**Comment by AngelFQC** [comment]
@LolaCapitani Do you have the configuration setting `$_configuration['send_two_inscription_confirmation_mail']` enabled?
If not, then try overriding `main/template/default/mail/content_registration_platform.tpl`

**Comment by AngelFQC** [comment]
Duplicated of #3203

**Comment by LolaCapitani** [comment]
Hi Angel!
Thank you for your answer...  : )
Maybe I didn't make myself clear. :'(

I tried your suggestion, although it was not exactly pertinent to my problem, but unfortunately nothing to do. 
Also I read the thread #3203 and it is not my problem.
So... what is my problem?

a) I modified the email templates in the path 'main\template\default\mail'. , for example the file 'new_user_first_email_confirmation.tpl'. I clicked on 'Proceed with cleanup'. I created a new user, the email that received the user is not the one with the template I modified

b) I followed the instructions written in the README.md file in the "overrides" folder (main/template/overrides).  And as I wrote before, nothing has changed. The email I received is always the one without my changes. Again, I clicked on "Proceed with cleanup".

Basically, I can't change the email templates in any way!

Thanks for your work!

Lola
(=^ ◡ ^=)

Desktop

OS: Linux Ubuntu 18.04 / Windows 10(64bit), Win 7(32 bit)
Browser CHROME 80.0.3987.87 , FIREFOX 78.01

Server

OS: Linux 3.10.0, LiteSpeed V7.6,
Version of Chamilo: 1.11.10
Version of PHP: PHP 7.1

**Comment by lonesomewalker** [comment]
Have you cleared the cache under:
/app/cache/twig
(delete the content in this folder)

?

**Comment by LolaCapitani** [comment]
Hello @lonesomewalker !
Thank you for your interest!

I followed your advice, I manually deleted all the content on the /app/cache/twig path.
The result is the same: nothing changes, new templates are not considered.

To be sure (once cleared the "cache" from the admin panel, and once cleared the contents of the /app/cache/twig folder), I tried either replacing the templates on the main/template/default/mail/ folder, or instead doing the recommended procedure, i.e. putting the new templates in the override folder (main/template/overrides), obviously inside a "mail" folder.
So the problem persists! (and on two different platforms).

I also did this shock test! :)

I deleted from the main/template/default/mail path the file new_user_first_email_confirmation.tpl. 
I created a new user and the user got the email with user email and password with the errors of the template even if the template doesn't exist anymore inside the path (main/template/default/mail/ folder) !

Thanks for your work!

Lola
(=^ ◡ ^=)

**Comment by lonesomewalker** [comment]
I just tried that with the latest release:
clone the whole mail directory into overrides.
Changed footer.
Registered.

Voila, change applied.

So, i suggest you have a look in your error.log.

**Comment by LolaCapitani** [comment]
Hi @lonesomewalker and @AngelFQC , and everybody!

I have documented my situation on video.

https://mega.nz/file/zYt2CIhL#G3Ta9wXPnP5Jy56km8As5ODvftmx9EgngEsXWIi1f94

This is the third time, with the third platform installed, that it is always the same problem.

I created a video to better explain the situation:

Here is a brief description:

I installed a virgin Chamilo platform.

I followed the main/template/default/mail path

I downloaded the file: new_user_firt_email_confirmation.tpl

I went back to the overrides folder and followed the instructions in the file README.md

I modified the file new_user_firt_email_confirmation.tpl

In the pat main/template/overrides/mail I inserted the file new_user_firt_email_confirmation.tpl

I created a new user.

The user received the subscription email but not the one I modified!


In the video you can see exactly all the steps I took.

So... Why can't I edit the email templates?

Thanks!

Lola
(=^ ◡ ^=)

**Comment by lonesomewalker** [comment]
Because you modified the wrong email template?

**Comment by LolaCapitani** [comment]
> 
> 
> Yes, but only if she modifies the right template ;-)



Hi @lonesomewalker
I learn now that there are _right_ and _wrong_ email templates. 
Could you tell me where I am wrong? 
I understand I'm not a genius, I'm just a normal user, I can't find this information in any manual, so I rely on the info I find on the forum and on the web.
And this is the result, I use the wrong template! Wow.
So which one is the right one? Could you kindly show me the way?
I am a volunteer teacher who, because of COVID, teaches distance learning to children who otherwise could not study.
I wanted in my school Chamilo instead of Moodle because it is european, it is "ours". 
And I was told that there was a community that would help me.
I spent hours making the video. The problem of email templates that do not change is also present in the forum, but nobody explains how to solve it, that's why I wrote here.
Thanks for your help

**Comment by LolaCapitani** [comment]
> 
> 
> When you modify a tpl file, you have to clear cache to see changes into Chamilo.

Hi @nosolored ,
if you read above, I have always cleared the cache of Chamilo, but without any result

Thanks for your help

**Comment by LolaCapitani** [comment]
> 
> 
> > @LolaCapitani Do you have the configuration setting `$_configuration['send_two_inscription_confirmation_mail']` enabled?
> > If not, then try overriding `main/template/default/mail/content_registration_platform.tpl`
> 
> Hi @LolaCapitani I reviewed the code of Chamilo and the `new_user_first_email_confirmation.tpl` template file is used when `send_two_inscription_confirmation_mail` is enabled or when `mail_template_system` is enabled.
> 
> If you are not using this configuration settings, please try overwriting the `main/template/default/mail/content_registration_platform.tpl` template file. Then clear the system cache.

@AngelFQC 

Thanks Angel,
I have a new test server, I've installed Chamilo again, now I'll proceed again with the advice you gave me and let you know. Previously it didn't work. But the platform was not "virgin". Now it is and I'll try again.
Thank you very much

Lola

**Comment by lonesomewalker** [comment]
@LolaCapitani you don't have to be a genius (we all here are also not "enlightened ;-) )
Just insert in ALL Templates some unique stuff and you see pretty fast which template is where.
This is common practice in situations where you don't know what happens when and you know you can't break something serious.

Called "trial and error", something which also teachers should be capable of ;-)

**Comment by ywarnier** [comment]
Hi @LolaCapitani 
Thanks for your efforts. We're the same as you, trying to find time for everything. Our public contribution here is voluntary too. E-mail templates are not considered a high priority because it's just not an essential part of online teaching, so sorry if we take a long time looking through this one and ask you for a lot of feedback.

Also, don't pay too much attention to @lonesomewalker 's direct language or short answers  :smile:. It's his style but that is not a representation of the whole community (and he's definitely helpful too).

There are many priorities for us to manage, so sometimes it takes a very long time to get to the bottom of a topic (you will also find that in the Moodle issue tracker, where some issues might take years to get solved, by the way).

**Comment by LolaCapitani** [comment]
Hi Yannick @ywarnier ,
Thank you for your kind answer, my humble participation in this "forum" is to help the community in a concrete way. Me and others will also join the association soon.

In essence me like others, I am doing the debugging. In this specific case,  me like others, I have noticed the email template overwriting malfunction.

If there is anything else I can do to help solve the problem, I am at disposal.

However, I must emphasize, given the words of lonesomewalker, that before posting any problem, I do extensive testing, I have colleagues do tests, and then if the problem persists, I report you the problem. 

I also thank you for the tone you used, very kind, because I think that in a community where people work (even for free) kindness and understanding is essential. :blush:

Lola
(=^ ◡ ^=)
